vividus-to-xray-exporter/src/main/java/org/vividus/xray/reader/JsonResourceReader.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2019-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.xray.reader;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.vividus.util.json.JsonUtils;
+
+public final class JsonResourceReader
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(JsonResourceReader.class);
+
+    private static final JsonUtils JSON_UTILS = new JsonUtils();
+
+    private JsonResourceReader()
+    {
+    }
+
+    public static List<FileEntry> readFrom(Path sourceDirectory) throws IOException
+    {
+        LOGGER.info("Starting to import resources from filesystem by path {}", sourceDirectory);
+
+        try (Stream<Path> paths = Files.walk(sourceDirectory))
+        {
+            return paths.map(Path::toFile)
+                        .filter(File::isFile)
+                        .map(f -> new FileEntry(f.getAbsolutePath(), readFileQuietly(f)))
+                        .filter(JsonResourceReader::filterJsonFile)
+                        .collect(Collectors.toList());
+        }
+    }
+
+    private static boolean filterJsonFile(FileEntry fileEntry)
+    {
+        boolean json = JSON_UTILS.isJson(fileEntry.getContent());
+        if (!json)
+        {
+            LOGGER.atInfo().addArgument(fileEntry::getPath).log("Content of file '{}' is not JSON");
+        }
+        return json;
+    }
+
+    private static String readFileQuietly(File file)
+    {
+        try
+        {
+            return FileUtils.readFileToString(file, StandardCharsets.UTF_8);
```

Comment:
`java.nio.file.Files#readString(java.nio.file.Path)` ?

Reply:
`java.nio.file.Files#readString(java.nio.file.Path)` fails if we try to read non-text file like pdf, we could
- expand JsonUtils.isJson to work with bytes
or
- add tika into class path to detect whether it's proper file